StereoKitC/xr_backends/openxr.cpp

```diff
@@ -697,6 +703,11 @@ void openxr_poll_events() {
 			break;
 		default: break;
 		}
+
```

Comment:
I'm curious if you considered putting this 'before' SK acts on the event? (my gut says 'after' is more correct, but I haven't gotten too deep here)

Also wonder what your thoughts are on having a system for consuming events, and not allowing them to fall through? I know some window event loops have this mechanism to say "I handled it already, you don't need to!", but I don't know if that'll actually be necessary here.

Reply:
Hmm so I haven't given this a _whole_ lot of thought, but one thing I considered is that allowing the user's callback to act before SK could cause this issue: since I am passing a reference of `event_buffer` to the user's callback via `&event_buff`, the user could modify (maybe inadvertently) the `event_buffer` struct data and cause issues when it's SK's turn to process the event. So that was my reasoning there - to avoid the user messing up the `event_buffer` data before SK got a chance to process it.

As for consuming an event and not allowing them to fall through, I guess it could be a useful feature for some folks? But I can't think of a particular scenario off the top of my head - which is probably just because I don't have much experience with OpenXR yet. But there could be good use cases! I just can't think of any.

From my perspective, I think I'd probably just be using the add callback feature to add event handlers for events that _aren't_ already handled by SK (like events coming from OpenXR extensions).  So the option to prevent fall through is a moot point to me. But I could definitely add the option if you'd like! Let lmk

Reply:
Okay, I think I'm satisfied with this for the moment! Leave it as it is, and we can make changes later if we find it necessary :)

Reply:
Okey doke!